spring-integration-zookeeper/src/test/java/org/springframework/integration/zookeeper/ZookeeperTestSupport.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright 2015 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.integration.zookeeper;
+
+import java.io.IOException;
+import java.util.concurrent.TimeUnit;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.curator.framework.CuratorFramework;
+import org.apache.curator.framework.CuratorFrameworkFactory;
+import org.apache.curator.retry.BoundedExponentialBackoffRetry;
+import org.apache.curator.test.TestingServer;
+import org.apache.curator.utils.CloseableUtils;
+import org.junit.After;
+import org.junit.Before;
+
+/**
+ * @author Marius Bogoevici
+ * @author Gary Russell
+ * @since 4.2
+ *
+ */
+public class ZookeeperTestSupport {
+
+	protected final Log log = LogFactory.getLog(this.getClass());
+
+	private TestingServer testingServer;
+
+	protected CuratorFramework client;
+
+	public ZookeeperTestSupport() {
+		super();
+	}
+
+	@Before
+	public void setUp() throws Exception {
+		testingServer = new TestingServer(true);
+		client = createNewClient();
+	}
+
+	@After
+	public void tearDown() {
+		closeClient(this.client);
+		try {
+			testingServer.stop();
+		}
+		catch (IOException e) {
+			log.warn("Exception thrown while shutting down ZooKeeper: ", e);
+		}
+		testingServer.getTempDirectory().delete();
+	}
+
+	protected CuratorFramework createNewClient() throws InterruptedException {
+		CuratorFramework client = CuratorFrameworkFactory.newClient(testingServer.getConnectString(),
+				new BoundedExponentialBackoffRetry(100, 1000, 3));
+		client.start();
+		client.blockUntilConnected(10000, TimeUnit.SECONDS);
+		return client;
+	}
+
+	protected void closeClient(CuratorFramework client) {
+		try {
+			CloseableUtils.closeQuietly(client);
+		}
+		catch (Exception e) {
+			log.warn("Exception thrown while closing client: ", e);
```

Comment:
Well, it's not exactly useless. We catch the exception is in order to be able to move on to properly shut down Zookeeper afterwards and remove the temporary files created by it. We might not need to do that if we were to use a shared Zookeeper instance.